pkg/util/podstore/podstore_test.go

```diff
@@ -0,0 +1,514 @@
+		name                    string
+		podStore                *PodStore
+		obj                     interface{}
+		expectedTimestampMapLen int
+	}{
+		{
+			name:                    "object is not Pod",
+			podStore:                podStore,
+			obj:                     node,
+			expectedTimestampMapLen: 0,
+		},
+		{
+			name:                    "Valid case",
+			obj:                     pod1,
+			podStore:                podStore,
+			expectedTimestampMapLen: 1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equal(t, 0, len(tt.podStore.timestampMap))
+			podStore.onPodCreate(tt.obj)
+			assert.Equal(t, tt.expectedTimestampMapLen, len(tt.podStore.timestampMap))
+			if tt.expectedTimestampMapLen > 0 {
+				assert.Equal(t, tt.podStore.clock.Now(), tt.podStore.timestampMap[string(tt.obj.(*v1.Pod).UID)].CreationTimestamp)
+			}
+		})
+	}
+}
+
+func Test_onPodDelete(t *testing.T) {
+	fakeClock := clock.NewFakeClock(refTime)
+	podStore := PodStore{
+		curPod:       nil,
+		prevPod:      cache.NewIndexer(podKeyFunc, cache.Indexers{PodIPIndex: podIPIndexFunc}),
+		podsToDelete: workqueue.NewDelayingQueueWithCustomClock(fakeClock, DeleteQueueName),
+		timestampMap: map[string]*LocalClock{"pod1": {CreationTimestamp: refTime}},
+		clock:        clock.NewFakeClock(refTime),
+	}
+	tests := []struct {
+		name                   string
+		obj                    interface{}
+		timeWait               time.Duration
+		expectedResultInPrev   *v1.Pod
+		expectedResultInDelete int
+	}{
+		{
+			name:                   "object is neither pod or DeletedFinalStateUnknown",
+			obj:                    node,
+			timeWait:               0,
+			expectedResultInPrev:   nil,
+			expectedResultInDelete: 0,
+		},
+		{
+			name:                   "Pod in prevPod and in podsToDelete",
+			obj:                    pod1,
+			timeWait:               5 * time.Minute,
+			expectedResultInPrev:   pod1,
+			expectedResultInDelete: 1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			expectedDeleteTime := refTime.Add(tt.timeWait)
+			podStore.onPodDelete(tt.obj)
+			prePods, _ := podStore.prevPod.ByIndex(PodIPIndex, "1.2.3.4")
+			if tt.expectedResultInPrev == nil {
+				assert.Equal(t, 0, len(prePods))
+			} else {
+				prePod := prePods[0].(*v1.Pod)
+				assert.Equal(t, tt.expectedResultInPrev, prePod)
+				assert.NotNil(t, podStore.timestampMap[string(prePod.UID)].DeletionTimestamp)
+
+				fakeClock.SetTime(expectedDeleteTime.Add(-10 * time.Millisecond))
+				err := wait.PollImmediate(10*time.Millisecond, 100*time.Millisecond, func() (bool, error) {
+					return podStore.podsToDelete.Len() == 0, nil
+				})
+				require.NoError(t, err, "Pod should not be added to PodsToDelete")
+				fakeClock.SetTime(expectedDeleteTime.Add(10 * time.Millisecond))
+				err = wait.PollImmediate(10*time.Millisecond, 100*time.Millisecond, func() (bool, error) {
+					return podStore.podsToDelete.Len() > 0, nil
+				})
```

Comment:
I think that 100 ms is small enough that this could sometimes fail in CI. You may want to use a larger timeout here (at least 1s) and possibly switch to `assert.Eventually`.

Reply:
Updated. Thanks.